src/overlays/actors/ovl_Bg_Hidan_Rock/z_bg_hidan_rock.c

```diff
@@ -47,31 +61,349 @@ static ColliderCylinderInit D_8088BF8C = {
+            this->unk_169 = 0;
+        }
+    }
+}
+
+void func_8088B90C(BgHidanRock* this, GlobalContext* globalCtx) {
+    if (Math_StepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y, 1.0f) != 0) {
+        this->actionFunc = func_8088B634;
+    }
+}
+
+void func_8088B954(BgHidanRock* this, GlobalContext* globalCtx) {
+    s16 unk_16A;
+
+    unk_16A = this->unk_16A;
+    if (unk_16A != 0) {
+        this->unk_16A = unk_16A - 1;
+        unk_16A = this->unk_16A;
+    }
+
+    if (unk_16A == 0) {
+        this->actionFunc = func_8088B79C;
+        this->dyna.actor.velocity.y = 0.0f;
+    }
+}
+
+void func_8088B990(BgHidanRock* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    this->unk_16A++;
+    if (this->dyna.unk_150 != 0.0f) {
+        this->dyna.actor.speedXZ = 0.0f;
+        player->stateFlags2 &= ~0x10;
+    }
+
+    if ((this->params == 0 && (Math_SmoothStepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y + 1820.0f,
+                                                  0.25f, 20.0f, 0.5f) < 0.1f)) ||
+        ((this->params != 0) && (Math_SmoothStepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y + 480.0,
+                                                    0.25f, 20.0f, 0.5f) < 0.1f))) {
+        if (this->params == 0) {
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BLOCK_BOUND);
+        }
+        this->unk_16A = 0x14;
+        this->actionFunc = func_8088B954;
+    }
+
+    this->unk_16C = (((this->dyna.actor.world.pos.y + 50.0f) - this->dyna.actor.home.pos.y) + 40.0f) / 80.0f;
+    if (this->params == 0) {
+        if (func_8004356C(&this->dyna) != 0) {
+            if (this->unk_169 == 0) {
+                this->unk_169 = 3;
+            }
+            Camera_ChangeSetting(globalCtx->cameraPtrs[0], 0x30);
+            return;
+        }
+        if (func_8004356C(&this->dyna) == 0) {
+            if (this->unk_169 != 0) {
+                Camera_ChangeSetting(globalCtx->cameraPtrs[0], 3);
+            }
+            this->unk_169 = 0;
+        }
+    }
+}
+
+void BgHidanRock_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanRock* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    if (this->actionFunc == func_8088B79C) {
+        Actor_MoveForward(&this->dyna.actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->dyna.actor, 0.0f, 0.0f, 0.0f, 4);
+    }
+
+    if (this->unk_16C > 0.0f) {
+        this->collider.dim.height = (D_8088BF8C.dim.height * this->unk_16C);
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Rock/func_8088B24C.s")
+void func_8088BC40(GlobalContext* globalCtx2, BgHidanRock* this) {
+    static const Gfx* D_8088BFC4[] = { 0x06012120, 0x060128A0, 0x06013020, 0x060137A0, 0x06013F20, 0x060146A0,
+                                       0x06014E20, 0x060155A0, 0x00000000, 0x00000000, 0x00000000 };
```

Comment:
```suggestion
    static Gfx* D_8088BFC4[] = { 0x06012120, 0x060128A0, 0x06013020, 0x060137A0, 0x06013F20, 0x060146A0,
                                       0x06014E20, 0x060155A0 };
```
These are probably padding. Also I don't think we do `const` like this on static variables unless it's needed to match.